src/commands/Minion/open.ts

```diff
@@ -53,36 +54,52 @@ export default class extends BotCommand {
 		return `You have ${itemsAvailable}.`;
 	}
 
-	async run(msg: KlasaMessage, [tier]: [string | undefined]) {
-		if (!tier) {
+	async run(msg: KlasaMessage, [quantity = 1, name]: [number, string | undefined]) {
+		if (!name) {
 			return msg.send(await this.showAvailable(msg));
 		}
 
-		const clueTier = ClueTiers.find(_tier => _tier.name.toLowerCase() === tier.toLowerCase());
-		if (!clueTier) {
-			return this.nonClueOpen(msg, tier);
+		const clue = ClueTiers.find(_tier => _tier.name.toLowerCase() === name.toLowerCase());
+		if (!clue) {
+			const osjsOpenable = Openables.find(openable =>
+				openable.aliases.some(alias => stringMatches(alias, name))
+			);
+			if (!osjsOpenable) {
+				return this.botOpenablesOpen(msg, quantity, name);
+			}
+			return this.osjsOpenablesOpen(msg, quantity, osjsOpenable);
 		}
+		return this.clueOpen(msg, quantity, clue);
+	}
 
-		const hasCasket = await msg.author.hasItem(clueTier.id);
-		if (!hasCasket) {
-			throw `You don't have any ${
+	async clueOpen(msg: KlasaMessage, quantity: number, clueTier: ClueTier) {
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		if ((userBank[clueTier.id] ?? 0) < quantity) {
```

Comment:
use `msg.author.numItemsInBankSync(clueTier.id) < quantity`, and the same for any similar checks

Reply:
whats wrong with this, other than the manual check if its not null